### ISSUE TITLE
Collapsable Lists to Player Info Screen

### ISSUE THREAD

**Issue by manitscold123** [endless-sky/endless-sky#3700]
Endless Sky has a lot of veteran players (and hackers, of course). But once you've played the game for 300+ hours, you end up with an insanely large networth and a fleet able to rival like, an archon.

But once you have as large of a fleet as I do, managing it in the Player Info screen becomes incredibly difficult, as you have swathes of ships to scroll through before you get to the few ships you want to park or unpark.

What I propose, to solve this, is to organize the Player [I]nfo tab by creating collapsable ships, so that the player can quickly and concisely organize their fleet. That, or a fleet organizer.

An alternative to this would be to sort the ship list in the Player Info tab every time you open it, in this specific order, from top to bottom
1. Player's current ship,
2. Disabled / Destroyed Escorts sorted with disabled then destroyed, sorted by ship model (or health)
3. Escorts in current system sorted by ship model
4. Escorts not parked but in other systems, sorted by ship model
5. Escorts parked
6. Fighters / Drones

This is a feature request, and I would like this to be a thing so that I (and my fellow micromanagers) can do our thing (micromanage).

**Comment by ghoulavenger** [comment]
The ships are not sorted individually. But you can drag and drop them into any order you like. This is in fact how you change your flagship. I'm not against your suggestion so to speak, but it also isn't necessary since manual sorting is quite possible. You can even move entire groups of ships at the same time (by selecting all of the ships you want to move with a shift click or a control click). And yes, I have a massive fleet, like 78 hurricanes and 150 deep rivers, plus my bactrian and a few boxwings. And thats just my active fleet, inactive I've got plenty of parked ships.

**Comment by Darcman99** [comment]
I'm all for this, however it think it wouldn't make it into the game simply because the game isn't about building up a massive fleet, but to really think about strategy of what ships you need for what purpose, and having to think about how to utilise them best in combat (escort commands).

I really support this idea though, having dropdown lists or at the least tags to assign to your ships would be a worthwhile and a longtime wanted feature.

**Comment by jafdy** [comment]
The focus of the game is on single ships and small fleets so this kind of feature is not a priority. If someone were to create it then it has a fair chance of merging but people already working on the game are unlikely to put the time in.

**Comment by MessyMix** [comment]
@Darcman99 : what @ghoulavenger suggested is already in the game. 

And you can also group escorts using control-(number) when a group of escorts is selected, so in the future you can provide commands to only that group of escorts—also a good feature for fleet management.

I'm sure this exact issue has come up in the past, and has been no'ed—the response is usually something along the lines of "this game is not intended for players to amass game-breakingly large fleets, so we will not accommodate for that."

**Comment by tehhowch** [comment]
There are some related issues to consider: 
https://github.com/endless-sky/endless-sky/issues/2769
https://github.com/endless-sky/endless-sky/issues/3201

**Comment by manitscold123** [comment]
@jafdy so you're saying that if I made a pr that organizes the player info fleet in the aforementioned order, it'll be considered?

**Comment by jafdy** [comment]
It would need to be an option and sensibly done, most likely you would need to add other sorting options too.

**Comment by ghoulavenger** [comment]
To do a sort well it would probably need programmable rules. But at the very least it would need different sorting options.

Options based on the following things:
1. Location
2. Type
3. Status (destroyed, disabled, parked, unparked)
4. Name
5. Groupping (I didn't know about the control groupping, but if it exists, ordering by groupping seems logical)

Now, given the complexity of all 5 options I listed, it makes sense to program the sort based on priority. Meaning that it can actually do 5 options, but in a specific order. Something like, first status, then location, then type, then groupping, then name. Sorting in such a way would give you unparked ships that are probably your fleet, organized by type and groupping while being alphabetically sorted. The rest of the list would largely show you where parked and disabled ships were while being conveniently groupped for readability. 

I don't think the programmable sort is necessary but it would be useful considering all of the sorting options. And I'll say again, I don't think the sort is necessary at all since you can manually manipulate the data easily enough -- it'd save some time, but only if the sort worked in the way you liked.

**Comment by Darcman99** [comment]
@MessyMix Yeah I see now how what i said is already in the game, I wasn't specific enough, I meant in the info screen to park/unpark all ships with the same tag.

**Comment by jafdy** [comment]
Calling us lazy is a very bad way of getting any help, the point we make is that it is not something we would do so you will need to do it yourself you convince someone to do it for you. We would happily provide pointers to help your proposal get merged . . . but that seems like too much effort now.

**Comment by KiLEdEnNis** [comment]
also about the laziness: there's various community members who spent a lot of their free time creating pull requests with new content, and said pull requests have been open and "ignored" for months, maybe even half a year
and then you act like this about a feature request, instead of an actual pull request, after it's been "ignored" for a day

seems uncalled for, so don't do it again please
